webapp/python/app.py

```diff
@@ -0,0 +1,504 @@
+
+
+# http://flask.pocoo.org/snippets/28/
+_paragraph_re = re.compile(r"(?:\r\n|\r|\n){2,}")
+
+
+@app.template_filter()
+@pass_eval_context
+def nl2br(eval_ctx, value):
+    result = "\n\n".join(
+        "<p>%s</p>" % p.replace("\n", "<br>\n")
+        for p in _paragraph_re.split(escape(value))
+    )
+    if eval_ctx.autoescape:
+        result = Markup(result)
+    return result
+
+
+# endpoints
+
+
+@app.route("/initialize")
+def get_initialize():
+    db_initialize()
+    return ""
+
+
+@app.route("/login")
+def get_login():
+    if get_session_user():
+        return flask.redirect("/")
+    return flask.render_template("login.html", me=None)
+
+
+@app.route("/login", methods=["POST"])
+def post_login():
+    if get_session_user():
+        return flask.redirect("/")
+
+    user = try_login(flask.request.form["account_name"], flask.request.form["password"])
+    if user:
+        flask.session["user"] = {"id": user["id"]}
+        flask.session["csrf_token"] = os.urandom(8).hex()
+        return flask.redirect("/")
+
+    flask.flash("アカウント名かパスワードが間違っています")
+    return flask.redirect("/login")
+
+
+@app.route("/register")
+def get_register():
+    if get_session_user():
+        return flask.redirect("/")
+    return flask.render_template("register.html", me=None)
+
+
+@app.route("/register", methods=["POST"])
+def post_register():
+    if get_session_user():
+        return flask.redirect("/")
+
+    account_name = flask.request.form["account_name"]
+    password = flask.request.form["password"]
+    if not validate_user(account_name, password):
+        flask.flash(
+            "アカウント名は3文字以上、パスワードは6文字以上である必要があります"
+        )
+        return flask.redirect("/register")
+
+    cursor = db().cursor()
+    cursor.execute("SELECT 1 FROM users WHERE `account_name` = %s", (account_name,))
+    user = cursor.fetchone()
+    if user:
+        flask.flash("アカウント名がすでに使われています")
+        return flask.redirect("/register")
+
+    query = "INSERT INTO `users` (`account_name`, `passhash`) VALUES (%s, %s)"
+    cursor.execute(query, (account_name, calculate_passhash(account_name, password)))
+
+    flask.session["user"] = {"id": cursor.lastrowid}
+    flask.session["csrf_token"] = os.urandom(8).hex()
+    return flask.redirect("/")
+
+
+@app.route("/logout")
+def get_logout():
+    flask.session.clear()
+    return flask.redirect("/")
+
+
+@app.route("/")
+def get_index():
+    me = get_session_user()
+
+    cursor = db().cursor()
+    cursor.execute(
+        "SELECT `id`, `user_id`, `body`, `created_at`, `mime` FROM `posts` ORDER BY `created_at` DESC"
+    )
+    posts = make_posts(cursor.fetchall())
+
+    return flask.render_template("index.html", posts=posts, me=me)
+
+
+@app.route("/@<account_name>")
+def get_user_list(account_name):
+    cursor = db().cursor()
+
+    cursor.execute(
+        "SELECT * FROM `users` WHERE `account_name` = %s AND `del_flg` = 0",
+        (account_name,),
+    )
+    user = cursor.fetchone()
+    if not user:
```

Comment:
列が存在しないことを期待する場合は明確にNoneで判定しても良いかなと思いました 🙏🏽 

```suggestion
    if user is None:
```